docs/docs/examples/example-basics/transport-example.md

```diff
@@ -0,0 +1,118 @@
+---
+sidebar_position: 2
+---
+
+# Network Example
+The undirected_graph implemented in `graaf::undirected_graph`
+The shortest path algorithm implemented in `graaf::algorithm::get_shortest_path`
+The graph traversal implemented in `graaf::algorithm::graph_traversal`
+
+Using the following graph:
+
+<pre>
+<p align="center">
+    <img src={require("/static/img/examples/example_usage.png").default}></img>
+</p>
+</pre>
+
+Custom vertex and edge. In order to use Dijkstra, we should provide the get_weight() function for the edge.
+
+```c++
+struct station {
+    int id{};
+	std::string name{};
+    double gaz_capacity{};
+};
+
+struct road {
+	int length{};
+    int get_weight() { return length; }
+};
+```
+
+# Initializing graph, start and end vertecies
+First, we create data structure and initializing graph with vertices and edges
+
+```c++
+struct graph_with_start_and_target {
+    graaf::undirected_graph<station, road> graph{};
+    graaf::vertex_id_t start{};
+    graaf::vertex_id_t target{};
+};
+
+graph_with_start_and_target create_graph_with_start_and_target() {
+...
+}
+```
+
+## Visualizing graph traversal result
+For shortest path, colouring edges with red to indicate the shortest path for both weighted and unweighted graph
+
+Result of unweighted shortest path, chosen edges are coloured red
+<pre>
+<p align="center">
+    <img src={require("/static/img/examples/example_shortest_path_unweighted.png").default}></img>
+</p>
+</pre>
+
+Result of weighted shortest path, chosen edges are coloured red
+<pre>
+<p align="center">
+    <img src={require("/static/img/examples/example_shortest_path_weighted.png").default}></img>
+</p>
+</pre>
+
+
+```c++
+void print_shortest_path(const graaf::undirected_graph<station, road>& graph,
+                const std::optional<graaf::algorithm::GraphPath<int>>& path, const std::string & filepath) {
+...
+}
+
+void print_visited_vertices(const graaf::undirected_graph<station, road>& graph,
+                            seen_vertices_t& seen,
+                            const std::string& filepath) {
+...
+}
+```
+
+Creating a vertex callback structure that will be passed as an argument in the graph traverse function
+The function is needed in order to be called inside the traverse function; see graph.tpp file lines 15 and 45 for context.
+
```

Comment:
[minor]: In general I would avoid directly referencing line numbers in source files as this is subjective to change quite often.
```suggestion

Creating a vertex callback structure that will be passed as an argument in the graph traverse function
The function is needed in order to be called inside the traverse function; see graph.tpp for context.

```